frontend/javascripts/admin/dataset/dataset_upload_view.js

```diff
@@ -446,6 +506,128 @@ class DatasetUploadView extends React.PureComponent<PropsWithFormAndRouter, Stat
   }
 }
 
+const baseStyle = {
+  flex: 1,
+  display: "flex",
+  flexDirection: "column",
+  alignItems: "center",
+  padding: "20px",
+  borderWidth: 2,
+  borderRadius: 2,
+  borderColor: "#eeeeee",
+  borderStyle: "dashed",
+  backgroundColor: "#fafafa",
+  color: "rgba(0, 0, 0, 0.85)",
+  fontSize: 16,
+  outline: "none",
+  transition: "border .24s ease-in-out",
+  cursor: "pointer",
+};
+
+const activeStyle = {
+  borderColor: "#2196f3",
+};
+
+const acceptStyle = {
+  borderColor: "#00e676",
+};
+
+const rejectStyle = {
+  borderColor: "#ff1744",
+};
+
+function FileUploadArea({ fileList, onChange }) {
+  const onDropAccepted = acceptedFiles => {
+    onChange(_.uniqBy(fileList.concat(acceptedFiles), file => file.path));
```

Comment:
In the admin_rest_api you did something like `file.path || file.webkitRelativePath || file.relativePath || file.name` to get a file identifier. Is `file.path` actually always defined?

Reply:
Good point. I dug a bit into this and it turns out that `react-dropzone` "enriches" the `File` object with the `path` attribute. See https://github.com/react-dropzone/file-selector/blob/1e23f600ade799c378cc0a65189708f68183cec8/src/file.ts#L18. Therefore, `path` should always exist. However, just to be on the safe side, I changed the code to `file.path || file.name` for both usages of `file.path` and added a comment which explains the rationale.

Reply:
Ok, I changed my mind a bit, since there are more occurences of `file.path` and I didn't want to include the fallback everywhere. I removed the `|| file.name` fallback completely now and added one check to the resumable-upload-creator which throws an exception in case `file.path` doesn't exist, anymore. Theoretically, this shouldn't happen, but since this "enrichment-feature" of react-dropzone seems to be very subtle (and not really documented), I wanted to integrate this into our code base somehow.